browser/js/signup/signup.js

```diff
@@ -19,15 +24,21 @@ app.controller('signupCtrl', function ($scope, AuthService, $state) {
 
         AuthService.signup(signupInfo)
         .then(function () {
-            return AuthService.login({email: signupInfo.email, password: signupInfo.password});
+            return AuthService.login({email: signupInfo.user.email, password: signupInfo.user.password});
         })
         .then(function () {
-            $state.go('home');
+            if (checkingOut) {
+                $state.go('checkout');
+            } else {
+                $state.go('home');
+            }
         })
         .catch(function () {
             $scope.error = 'Invalid signup credentials.';
         });
 
     };
 
+    $scope.states = ["", "AK", "AL", "AR", "AZ", "CA", "CO", "CT", "DC", "DE", "FL", "GA", "HI", "IA", "ID", "IL", "IN", "KS", "KY", "LA", "MA", "MD", "ME", "MI", "MN", "MO", "MS", "MT", "NC", "ND", "NE", "NH", "NJ", "NM", "NV", "NY", "OH", "OK", "OR", "PA", "RI", "SC", "SD", "TN", "TX", "UT", "VA", "VT", "WA", "WI", "WV", "WY"];
+
```

Comment:
Maybe we can put $scope.states on the rootscope, so that it can be used for the address-form directive.